File: generations_calculation_function.R
##generations calculation function

gencalc = function(time){
  roundedTime = round(time)
  if(roundedTime == 0){noSteps = 1}
  else{
    subDens = as.numeric(densities[roundedTime,])
    subDensStack = subDens/sum(subDens)
    noSteps = sample(1:ncol(densities), 1, prob = subDensStack)
  }
  
  return(noSteps)
  
}

File: code/generations_calculation_function.jl
using Distributions
function GenCalc(time::Float64, densities::DataFrame)
    subDens = Vector(densities[ceil(Int, time),:])
    subDensStack = subDens/sum(subDens)
    noSteps = wsample(1:ncol(densities), subDensStack, 1)
    return(noSteps)
end